### PR TITLE
disttask: fix can't reset service_scope

### LINKED ISSUE #50104: `tidb_service_scope` not works for Distributed eXecution Framework
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. deploy cluster with 3 tidb nodes
2. connect to tidb-0 and tidb-2 execute sql ` set global tidb_service_scope = "background";`
3. check the `mysql.dist_framework_meta`
4. Import data

### 2. What did you expect to see? (Required)
only `background` tidb nodes works for import data 
### 3. What did you see instead (Required)
all the tidb nodes running for import data
```sql
[11:13:58]TiDB root:(none)> set global tidb_service_scope = "background";
Query OK, 0 rows affected
[11:14:00]TiDB root:(none)> select * from mysql.dist_framework_meta;
+---------------------------------------------------------------------------+------+-----------+-------------+
| host                                                                      | role | cpu_count | keyspace_id |
+---------------------------------------------------------------------------+------+-----------+-------------+
| tc-tidb-1.tc-tidb-peer.testbed-global-operator-tps-5550170-1-849.svc:4000 |      | 16        | -1          |
| tc-tidb-0.tc-tidb-peer.testbed-global-operator-tps-5550170-1-849.svc:4000 |      | 16        | -1          |
| tc-tidb-2.tc-tidb-peer.testbed-global-operator-tps-5550170-1-849.svc:4000 |      | 16        | -1          |
+---------------------------------------------------------------------------+------+-----------+-------------+

11:18:03]TiDB root:test> select id, task_key, type, state, step, exec_id, cast(error as char) from mysql.tidb_background_subtask where task_key = 1;
+----+----------+------+---------+------+---------------------------------------------------------------------------+---------------------+
| id | task_key | type | state   | step | exec_id                                                                   | cast(error as char) |
+----+----------+------+---------+------+---------------------------------------------------------------------------+---------------------+
| 1  | 1        | 2    | running | 3    | tc-tidb-0.tc-tidb-peer.testbed-global-operator-tps-5550170-1-849.svc:4000 | <null>              |
| 2  | 1        | 2    | running | 3    | tc-tidb-1.tc-tidb-peer.testbed-global-operator-tps-5550170-1-849.svc:4000 | <null>              |
| 3  | 1        | 2    | running | 3    | tc-tidb-2.tc-tidb-peer.testbed-global-operator-tps-5550170-1-849.svc:4000 | <null>              |
+----+----------+------+---------+------+---------------------------------------------------------------------------+---------------------+
```
### 4. What is your TiDB version? (Required)
```sql
[11:16:16]TiDB root:(none)> select tidb_version();
+-----------------------------------------------------------+
| tidb_version()                                            |
+-----------------------------------------------------------+
| Release Version: v7.6.0-alpha                             |
| Edition: Community                                        |
| Git Commit Hash: 62c83d40fe2ed742bfaf4b7c64eca0bb1853a94d |
| Git Branch: heads/refs/tags/v7.6.0-alpha                  |
| UTC Build Time: 2024-01-04 14:31:13                       |
| GoVersion: go1.21.5                                       |
| Race Enabled: false                                       |
| Check Table Before Drop: false                            |
| Store: tikv                                               |
+-----------------------------------------------------------+
```

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: close #50104 

Problem Summary:
set global tidb_service_scope not take effect.
StartManager can't update role since task executor manager has already insert a row in dist_framework_meta
### What changed and how does it work?
When update row in dist_framework_meta, update role.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/disttask/framework/storage/task_table.go [hunk 1]
```diff
@@ -690,14 +690,15 @@ func (stm *TaskManager) StartManager(ctx context.Context, tidbID string, role st
...
+// if the record exists, update the cpu_count and role.
 func (*TaskManager) StartManagerSession(ctx context.Context, se sessionctx.Context, execID string, role string) error {
 	cpuCount := cpu.GetCPUCount()
 	_, err := sqlexec.ExecSQL(ctx, se, `
 		insert into mysql.dist_framework_meta(host, role, cpu_count, keyspace_id)
 		values (%?, %?, %?, -1)
-		on duplicate key update cpu_count = %?`,
-		execID, role, cpuCount, cpuCount)
+		on duplicate key
+		update cpu_count = %?, role = %?`,
```
Comment (D3Hunter) [inline pkg/disttask/framework/storage/task_table.go:700]: we'll call it periodically in taskexecutor manager, will override it

we need another interface to change/update role

Comment (ywqzzy) [reply pkg/disttask/framework/storage/task_table.go:700]: Won't override it, since we have store the role in global memory

Comment (ywqzzy) [reply pkg/disttask/framework/storage/task_table.go:700]: If the tidb crash or exit, it's safe to override it.

Comment (D3Hunter) [reply pkg/disttask/framework/storage/task_table.go:700]: suppose we `set` before `initMeta` buf after get get `config.GetGlobalConfig().Instance.TiDBServiceScope`, the scope will be wrong for 90s

Comment (ywqzzy) [reply pkg/disttask/framework/storage/task_table.go:700]: There are 2 timeflow:
1. 
a. config.GetGlobalConfig().Instance.TiDBServiceScope = valStr
b. startManager in recoverLoop
c. startManager in set.go

2. 
a. config.GetGlobalConfig().Instance.TiDBServiceScope = valStr
b. startManager in set.go
c. startManager in recoverLoop


both of them will use the latest one.

Comment (Benjamin2037) [reply pkg/disttask/framework/storage/task_table.go:700]: If the above scenario can be written a case to cover it？

**Review by Benjamin2037 — APPROVED** [review]
LGTM

File: pkg/disttask/framework/taskexecutor/manager.go [hunk 2]
```diff
@@ -132,10 +135,24 @@ func (m *Manager) initMeta() (err error) {
...
+	for i := 0; i < retrySQLTimes; i++ {
+		err = m.taskTable.RecoverMeta(m.ctx, m.id, config.GetGlobalConfig().Instance.TiDBServiceScope)
+		if err == nil {
+			break
+		}
+		if err1 := m.ctx.Err(); err1 != nil {
+			return err1
+		}
+		if i%10 == 0 {
+			logutil.Logger(m.logCtx).Warn("start manager failed",
```
Comment (D3Hunter) [inline pkg/disttask/framework/taskexecutor/manager.go:148]:
```suggestion
			logutil.Logger(m.logCtx).Warn("recover meta failed",
```

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**